base/verify_install_node

```diff
@@ -0,0 +1,173 @@
+# if node verification fails
+if [ "${5}" == "FAIL_ON_ERROR" ]; then
+  verify_failure_action="fail"
+elif [ "${5}" == "WARN_ON_ERROR" ]; then
+  verify_failure_action="warn"
+else
+  echo "Either FAIL_ON_ERROR or WARN_ON_ERROR must be specified."
+  exit 1
+fi
+
+# The temporary location where the downloaded files are stored while being
+# verified.  The script will ensure that this directory is removed before 
+# the script exits.
+script_name=`basename ${0}`
+tmp_dir=`mktemp -d /tmp/${script_name}.XXXXXX` || exit 1
+
+# Used to issue warnings to the user, directing the message to standard error
+function warn {
+  echo ${1} 1>&2;
+}
+
+# Used to cleanup any temporary directories
+function cleanUp {
+  rm -Rf "${tmp_dir}"
+}
+
+# Used to cleanup any temporary directories and immediately exit this
+# script with an error code indicating failure
+function exitAsFailure {
+  cleanUp
+  exit 1
+}
+
+# Used to handle not being able to download the checksum file needed to
+# verify the Node.js binary by either causing the script to fail or just
+# print a warning message depending on the parameters given to this script.
+function handleFailedChecksumDownload {
+  warn "The files required to verify the requested Node.js binary could not be"
+  warn "downloaded."
+  warn ""
+
+  if [ "${verify_failure_action}" == "fail" ]; then
+    warn "Aborting the installation."
+    warn ""
+    exitAsFailure
+  elif [ "${verify_failure_action}" == "warn" ]; then
+    warn "The installation will continue, but it is strongly recommended"
+    warn "that you install a version of Node.js that can been verified."
+    warn ""
+  else
+    warn "Unable to determine the appropriate action to take if"
+    warn "verifying the Node.js binary cannot proceed."
+    warn ""
+    warn "Continuing with the installation."
+    warn ""
+  fi
+}
+
+# Used to handle a failure in verifying the specified Node.js binary
+# by either causing the script to fail or just print a warning message
+# depending on the parameters given to this script.
+function handleFailedVerification {
+  warn "The Node.js binary could not be verified."
+  warn "This means it may not be an officially released Node.js binary"
+  warn "or may have been tampered with."
+  warn ""
+
+  if [ "${verify_failure_action}" == "fail" ]; then
+    warn "Aborting the installation."
+    warn ""
+    exitAsFailure
+  elif [ "${verify_failure_action}" == "warn" ]; then
+    warn "The installation will continue, but it is strongly recommended"
+    warn "that you install a version of Node.js that has been verified."
+    warn ""
+  else
+    warn "Unable to determine the appropriate action to take if"
+    warn "verifying the Node.js binary failed."
+    warn ""
+    warn "Continuing with the installation."
+    warn ""
+  fi
+}
+
+# Make sure we are in the correct place in the fs and it is pristine
+cleanUp
+mkdir -p "${tmp_dir}"
+cd "${tmp_dir}"
+
+# wget is used over curl so that a non-zero exit code is produced if a 
```

Comment:
I didn't know `curl` has a `--fail` option.  I will look into using it so that the image doesn't have a dependency on `wget`.